gusto/diagnostics.py

```diff
@@ -1581,3 +1582,34 @@ def setup(self, domain, state_fields):
             state_fields (:class:`StateFields`): the model's field container.
         """
         super().setup(domain, state_fields, vorticity_type="relative")
+
+
+class TracerDensity(DiagnosticField):
+    """Diagnostic for computing the density of a tracer. This is
+    computed as the product of a mixing ratio and dry density"""
+
+    name = "TracerDensity"
+
+    def __init__(self, m_X, rho_d):
+        """
+        Args:
+            m_X: the mixing ratio of the tracer
+            rho_d: the dry density of the tracer
+        """
+        super().__init__(method='interpolate', required_fields=(m_X, rho_d))
+        self.m_X = m_X
+        self.rho_d = rho_d
+
+    def setup(self, domain, state_fields):
+        """
+        Sets up the :class:`Function` for the diagnostic field.
+
+        Args:
+            domain (:class:`Domain`): the model's domain object.
+            state_fields (:class:`StateFields`): the model's field container.
+        """
+        m_X = state_fields(self.m_X)
+        rho_d = state_fields(self.rho_d)
+        space = m_X.function_space()
+        self.expr = m_X*rho_d
+        super().setup(domain, state_fields, space=space)
```

Comment:
If we include `space` as an argument to the `__init__` then we won't need to include it here